### PR TITLE
Current progress of octod implementation

### PR DESCRIPTION
**Pull request by mihails-strasuns-sociomantic**
- core connection module
- example script that was used to create our D2 progress tracking project
- higher level API wrappers as required for that script

### CODE REVIEW

File: dub.sdl [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name "dghapi"
```
Comment (gavin-norman-sociomantic) [inline dub.sdl]: octod ;)

File: source/octod/core.d [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+    /// If present, will be used as auth username
+    string username;
+    /// If 'this.username' is present, will be used as auth password
+    string password;
+    /// If 'this.username' is empty, will be used as auth token
+    string oauthToken;
+    /// Sent as 'Accept' header
+    string accept = "application/vnd.github.v3+json";
+    /// By default client works in dry run mode
+    bool dryRun = true;
```
Comment (leandro-lucarella-sociomantic) [inline source/octod/core.d:37]: Really? :thinking:

Comment (mihails-strasuns-sociomantic) [reply source/octod/core.d:37]: Safety precaution ;)

File: dub.sdl (hunk 1 above)
Comment (mihails-strasuns-sociomantic) [reply dub.sdl]: oops!

**Comment by mihails-strasuns-sociomantic** [comment]
Names fixed

**Comment by leandro-lucarella-sociomantic** [comment]
Let's go!

**Pull request merged**